lib/vtls/openssl.c

```diff
@@ -3025,6 +3081,15 @@ static CURLcode ossl_connect_step1(struct Curl_easy *data,
       infof(data, "error importing windows ca store, continuing anyway\n");
   }
 #endif
+  if(ca_file_pem && load_cacert_from_memory(backend->ctx, ca_file_pem)) {
```

Comment:
`load_cacert_from_memory()` returns a CURLcode but here you ignore that and just assumes the error is always due to badfile. Isn't it better to use the passed back return code?

Reply:
I rewrote this line and return the result of load_cacert_from_memory